misk-testing/src/main/kotlin/misk/tracing/MockTracingModule.kt

```diff
@@ -0,0 +1,33 @@
+package misk.tracing
+
+import com.google.inject.Provides
+import io.opentracing.Tracer
+import io.opentracing.mock.MockTracer
+import io.opentracing.util.GlobalTracer
+import misk.inject.KAbstractModule
+import javax.inject.Singleton
+
+
+class MockTracingModule : KAbstractModule() {
+  companion object {
+    val tracer : MockTracer = MockTracer()
```

Comment:
having a global seems dangerous for testing

Reply:
Yeah the challenge is that the underlying GlobalTracer tracer is already a global. I can at least keep this as an instance

Reply:
eh we kind of need to keep it global because of the GlobalTracer; otherwise different test cases end up getting different versions of the tracer

Reply:
that seems like what you want? a new tracer for every test case

Reply:
imo `@Inject` the `Tracer` vs relying on `GlobalTracer`

Reply:
Yeah, but you can't override the GlobalTracer's tracer once it's set

Reply:
Hmm good idea re: injecting, I'll look into that